Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -0,0 +1,206 @@
+//
+//  Copyright (C) 2020 Shrey Aryan
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#ifndef GENERAL_FILE_READER_H
+#define GENERAL_FILE_READER_H
+#include <RDGeneral/BadFileException.h>
+#include <RDStreams/streams.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "MolSupplier.h"
+
+namespace RDKit {
+namespace GeneralMolSupplier {
+struct SupplierOptions {
+  bool takeOwnership = true;
+  bool sanitize = true;
+  bool removeHs = true;
+  bool strictParsing = true;
+
+  std::string delimiter = "\t";
+  int smilesColumn = 0;
+  int nameColumn = 1;
+  bool titleLine = true;
+
+  std::string nameRecord = "";
+  int confId2D = -1;
+  int confId3D = 0;
+};
+//! current supported file formats
+std::vector<std::string> supportedFileFormats{"sdf", "mae", "maegz", "smi",
+                                              "csv", "txt", "tsv",   "tdt"};
+//! current supported compression formats
+std::vector<std::string> supportedCompressionFormats{"gz"};
+
+//! determines whether file and compression format are valid
+bool validate(std::string& fileFormat, std::string& compressionFormat) {
+  bool flag_fileFormat = true;
+  bool flag_compressionFormat = true;
+
+  //! Case 1: No compression format
+  if (compressionFormat.empty()) {
+    //! Unconventional format types
+    if (fileFormat.compare("maegz") == 0) {
+      fileFormat = "mae";
+      compressionFormat = "gz";
+      return true;
+    }
+
+    //! Usual file formats
+    return std::find(supportedFileFormats.begin(), supportedFileFormats.end(),
+                     fileFormat) != supportedFileFormats.end();
+  }
+  //! Case 2: With compression format
+  else {
+    flag_fileFormat =
+        std::find(supportedFileFormats.begin(), supportedFileFormats.end(),
+                  fileFormat) != supportedFileFormats.end();
+    flag_compressionFormat =
+        std::find(supportedCompressionFormats.begin(),
+                  supportedCompressionFormats.end(),
+                  compressionFormat) != supportedCompressionFormats.end();
+
+    //! Case A: Valid file format and valid compression format
+    if (flag_fileFormat && flag_compressionFormat) return true;
```

Comment:
This is a style thing, so a matter of debate, but I prefer to use braces round even single line conditions.  I think it looks tidier, but more importantly they protect you from adding something else under the condition (debugging writes being a classic) and forgetting that the 2nd line won't be executed in the condition.  Your IDE should protect you from this, but it doesn't always.  I believe the SSL Heartbleed bug was due to something like that.

Reply:
I should have caught that in my review too. We now put brackets around things like this in the RDKit code. It's not incredibly important, but it's nice to remain consistent

Unfortunately it's not something that clang-format will do for you (you have to use clang-tidy for this).